core/sail/base/src/main/java/org/eclipse/rdf4j/sail/inferencer/InferencerConnectionWrapper.java

```diff
@@ -93,7 +98,12 @@ public void prepare() throws SailException {
 	@Override
 	public void commit() throws SailException {
 		flushUpdates();
+		long count = getWrappedConnection().getStatements(null, null, null, true).stream().distinct().count();
 		super.commit();
+		long count2 = getWrappedConnection().getStatements(null, null, null, true).stream().distinct().count();
+		if (count != count2) {
+			System.out.println(count + " != " + count2);
+		}
 	}
```

Comment:
What I've found out is that the count is correct before calling `super.commit()` but wrong after.